WS2012R2/lisa/setupscripts/AddVhdxHardDisk.ps1

```diff
@@ -358,16 +358,11 @@ if ($testParams -eq $null -or $testParams.Length -lt 3)
     return $False
 }
 
-# WS 2012, 2008 R2 do not support generation 2 VMs
-$OSInfo = get-wmiobject Win32_OperatingSystem -ComputerName $hvServer
-if ( ($OSInfo.Caption -match '.2008 R2.') -or 
-     ($OSInfo.Caption -match '.2012 [^R2].')) {
+$vmGeneration = Get-VM $vmName -ComputerName $hvServer| select -ExpandProperty Generation -ErrorAction SilentlyContinue
```

Comment:
We need to keep the OSInfo check as 2012 and 2008R2 do not have the 'Generation' property in PowerShell.